lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9405,8 +9422,12 @@ ClangImporter::Implementation::importDeclForDeclContext(
 
   // There's a cycle. Is the declaration imported enough to break the cycle
   // gracefully? If so, we'll have it in the decl cache.
-  if (auto cached = importDeclCached(contextDecl, version, useCanonicalDecl))
-    return cached;
+  auto cached = importDeclCached(contextDecl, version, useCanonicalDecl);
+  if (cached.hasValue() && cached.getValue()) {
+    return cached.getValue();
+  } else if (cached.hasValue()) {
```

Comment:
same

Reply:
`ClangImporter/attr-swift_name.swift` fails if we return null from here because the diagnostics code for swift_name_circular_context_import doesn't get hit to append the warning+notes:

```
/Users/plotfi/local/S/swift/test/ClangImporter/Inputs/custom-modules/ObjCIRExtras.h:78:52: warning: cycle detected while resolving 'MutuallyCircularNameA' in swift_name attribute for 'MutuallyCircularNameB'
SWIFT_NAME(MutuallyCircularNameA.Inner) @interface MutuallyCircularNameB : NSObject @end

/Users/plotfi/local/S/swift/test/ClangImporter/Inputs/custom-modules/ObjCIRExtras.h:77:52: note: while resolving 'MutuallyCircularNameB' in swift_name attribute for 'MutuallyCircularNameA'
SWIFT_NAME(MutuallyCircularNameB.Inner) @interface MutuallyCircularNameA : NSObject @end

/Users/plotfi/local/S/swift/test/ClangImporter/Inputs/custom-modules/ObjCIRExtras.h:78:52: note: please report this issue to the owners of 'ObjCIRExtras'
SWIFT_NAME(MutuallyCircularNameA.Inner) @interface MutuallyCircularNameB : NSObject @end
```

Since this function will return nullptr from this point regardless, do we want to bail before emitting these warnings?